source/fundamentals/crud/read-operations/cursor.txt

```diff
@@ -1,8 +1,156 @@
+.. _cursor-golang:
+
 =========================
 Access Data From a Cursor
 =========================
 
 .. default-domain:: mongodb
 
-Read operations that return multiple documents do not immediately return
-all values matching the query.
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to access data with a **cursor**.
+
+A cursor is a mechanism that allows an application to iterate over
+database results while holding only a subset of them in memory at a
+given time. Read operations that match multiple documents use a cursor
+to return those documents in batches as opposed to all at once.
+
+Sample Cursor
+~~~~~~~~~~~~~
+
+Each section uses the following ``cursor`` variable, which is a
+``Cursor`` struct that contains all the documents in a collection:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin find
+   :end-before: end find
+
+.. important::
+
+   A cursor is not goroutine safe. Do not use the same cursor in
+   multiple goroutines at the same time.
+
+.. _individual-documents-golang:
+
+Retrieve Documents Individually
+-------------------------------
+
+To retrieve documents from your cursor individually, use the
+``Next()`` method.
+
+This method blocks the current goroutine. That means the function
+returns a document if each of the following is true:
+
+- A document is currently or will later be available
+- No errors occurred
+- The ``MongoClient`` didn't exhaust the cursor
+- The context didn't expire
+
+.. include:: /includes/fundamentals/exhausted-cursor-definition.rst
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin cursor next
+   :end-before: end cursor next
+
+Documents Only in the Cursor
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To attempt retrieving a document from your cursor, use the ``TryNext()``
+method.
+
+This method is unable to block the current goroutine. That means the
+function returns a document if each of the following is true:
+
+- A document is currently available
+- No errors occurred
+- The ``MongoClient`` didn't exhaust the cursor
+- The context didn't expire
+
+.. include:: /includes/fundamentals/exhausted-cursor-definition.rst
```

Comment:
Suggestion: As admonitions are less likely to be read by readers, could this definition be removed from an admonition and placed in the main text?

Suggestion: Could this information only be presented 1 time on the page? I believe this can be defined once in the "Retrieve Documents Individually" section as it is the parent section to all sections that contain this admonition.

Reply:
After discussing this section with NL, we decided to remove the "exhaust the cursor" bullet since by definition, thats the first bullet (where it says IF a document exists). I'll also be removing this note since there is no need to define it anymore.